### PR TITLE
Feature/sent on report

### PR DESCRIPTION
**Pull request by SumeN**
For PBI 73322

### CODE REVIEW

File: src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/GetAllAatfSentOnDataCsvTests.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+                var dataB2B = from x in results.Tables[0].AsEnumerable()
+                              where x.Field<string>("Name of AATF") == aatf.Name && x.Field<string>("Obligation") == "B2B"
+                              select x;
+                dataB2B.AsQueryable().Count().Should().Be(14);
+
+                var dataB2C = from x in results.Tables[0].AsEnumerable()
+                              where x.Field<string>("Name of AATF") == aatf.Name && x.Field<string>("Obligation") == "B2C"
+                              select x;
+                dataB2C.AsQueryable().Count().Should().Be(14);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/GetAllAatfSentOnDataCsvTests.cs]: Can some tests be added to test the stored procedure parameters

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AatfReports/GetAllAatfSentOnDataCsvHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        [Fact]
+        public async Task GetAllAatfSentOnDataCsvHandler_StringParameters_ReturnsFileName()
+        {
+            var authorization = new AuthorizationBuilder().AllowInternalAreaAccess().Build();
+            var context = A.Fake<WeeeContext>();
+            var commanDataAccess = A.Fake<ICommonDataAccess>();
+            var csvWriterFactory = A.Fake<CsvWriterFactory>();
+            int complianceYear = 2019;
+            string obligationType = "B2C";
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AatfReports/GetAllAatfSentOnDataCsvHandlerTests.cs]: Can we add test for when some of the parameters are null, probably easier to split it out into test for parameter combinations

File: src/EA.Weee.RequestHandlers/Admin/AatfReports/GetAllAatfSentOnDataCsvHandler.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            if (request.ObligationType != null)
+            {
+                fileName += "_" + request.ObligationType;
+            }
+
+            fileName += string.Format("_AATF WEEE sent on for treatment_{0:ddMMyyyy_HHmm}.csv",
+                                DateTime.UtcNow);
+
+            string fileContent = string.Empty;
+            if (obligatedData.Tables.Count > 0)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/AatfReports/GetAllAatfSentOnDataCsvHandler.cs]: Add test that if no are no tables that the file content should be string.empty

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs [hunk 4]
```diff
@@ -675,5 +676,147 @@ public async Task PostAatfObligatedData_Always_SetsInternalBreadcrumbToViewRepor
...
+        /// <returns></returns>
+        [Fact]
+        public async Task PostAatfSentOnData_Always_SetsInternalBreadcrumbToViewReports()
+        {
+            // Act
+            await controller.AatfSentOnData(A.Dummy<AatfSentOnDataViewModel>());
+
+            // Assert
+            Assert.Equal("View reports", breadcrumb.InternalActivity);
+        }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs]: Add test for DownloadAatfSentOnDataCsv to ensure the request is called

File: src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/GetAllAatfSentOnDataCsvTests.cs (hunk 1 above)
Comment (SumeN) [reply src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/GetAllAatfSentOnDataCsvTests.cs]: Added

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AatfReports/GetAllAatfSentOnDataCsvHandlerTests.cs (hunk 2 above)
Comment (SumeN) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AatfReports/GetAllAatfSentOnDataCsvHandlerTests.cs]: Added

File: src/EA.Weee.RequestHandlers/Admin/AatfReports/GetAllAatfSentOnDataCsvHandler.cs (hunk 3 above)
Comment (SumeN) [reply src/EA.Weee.RequestHandlers/Admin/AatfReports/GetAllAatfSentOnDataCsvHandler.cs]: Added test

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs (hunk 4 above)
Comment (SumeN) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs]: Added test
